pkg/operation/common/managedresources_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common_test
+
+import (
+	"context"
+	"fmt"
+
+	mockclient "github.com/gardener/gardener/pkg/mock/controller-runtime/client"
+	. "github.com/gardener/gardener/pkg/operation/common"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	resourcesv1alpha1 "github.com/gardener/gardener-resource-manager/pkg/apis/resources/v1alpha1"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+var _ = Describe("ManagedResources", func() {
+	var (
+		ctrl *gomock.Controller
+		c    *mockclient.MockClient
+
+		ctx         = context.TODO()
+		fakeErr     = fmt.Errorf("fake err")
+		name        = "foo"
+		namespace   = "bar"
+		keepObjects = true
+		data        = map[string][]byte{"some": []byte("data")}
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		c = mockclient.NewMockClient(ctrl)
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	Describe("#DeployManagedResource", func() {
+		It("should return the error of the secret reconciliation", func() {
```

Comment:
Can you also add tests for the case, that the secret/mr does not already exist?

Reply:
OK, now I get what you mean above (at least the first part). I haven't added these "cases" intentionally as they don't belong to these functions here, IMO. They belong to unit tests for `CreateOrUpdate`, but these unit tests are about higher-level functions that call `CreateOrUpdate` internally. If we always multiply them then we will blow up our tests artificially without a lot of benefit, or? I can add them for sure, but where do you suggest to pull the line?

Reply:
Good point, agreed. What is tested now, is probably already enough.